server/src/main/scala/com/cloudera/livy/utils/LivySparkUtils.scala

```diff
@@ -112,4 +132,23 @@ object LivySparkUtils {
         throw new IllegalArgumentException(s"Fail to parse Spark version from $version")
     }
   }
+
+  /**
+   * Return Scala binary version, if it cannot be parsed from input version string, it will
+   * pick default Scala version related to Spark version.
+   *
+   * @param version Scala binary version String
+   * @param sparkVersion formatted Spark version.
+   * @return Scala binary version String based on Spark version and livy conf.
+   */
+  def formatScalaVersion(version: String, sparkVersion: (Int, Int)): String = {
```

Comment:
And how about change it to formatScalaVersion(scalaVersion:String, sparkVersion:String) ?
formatSparkVersion can be called inside this method.